core/src/main/java/org/mule/runtime/core/processor/strategy/SynchronousProcessingStrategyFactory.java

```diff
@@ -45,4 +59,71 @@ public ProcessingStrategy create(MuleContext muleContext, String schedulersNameP
     return SYNCHRONOUS_PROCESSING_STRATEGY_INSTANCE;
   }
 
+  static class SinkPerThreadSink implements Sink, Disposable {
+
+    private Supplier<Sink> sinkSupplier;
+    private Cache<Thread, Sink> sinkCache =
+        newBuilder().weakValues().removalListener(notification -> disposeIfNeeded(notification.getValue(), NOP_LOGGER)).build();
```

Comment:
Because i'm not interested in logging the failure to dispose a sink during garbage collection.